src/e2e/java/teammates/e2e/cases/sql/FeedbackMsqQuestionE2ETest.java

```diff
@@ -0,0 +1,150 @@
+
+        instructor = testData.instructors.get("instructor");
+        course = testData.courses.get("course");
+        feedbackSession = testData.feedbackSessions.get("openSession");
+        student = testData.students.get("alice.tmms@FMsqQn.CS2104");
+    }
+
+    @Test
+    @Override
+    public void testAll() {
+        testEditPage();
+        logout();
+        testSubmitPage();
+    }
+
+    @Override
+    protected void testEditPage() {
+        InstructorFeedbackEditPage feedbackEditPage = loginToFeedbackEditPage();
+
+        ______TS("verify loaded question");
+        FeedbackQuestion loadedQuestion = testData.feedbackQuestions.get("qn1ForFirstSession");
+        FeedbackMsqQuestionDetails questionDetails = (FeedbackMsqQuestionDetails) loadedQuestion
+                .getQuestionDetailsCopy();
+        feedbackEditPage.verifyMsqQuestionDetails(1, questionDetails);
+
+        ______TS("add new question");
+        // add new question exactly like loaded question
+        loadedQuestion.setQuestionNumber(2);
+        feedbackEditPage.addMsqQuestion(loadedQuestion);
+
+        feedbackEditPage.verifyMsqQuestionDetails(2, questionDetails);
+        verifyPresentInDatabase(loadedQuestion);
+
+        ______TS("copy question");
+        FeedbackQuestion copiedQuestion = testData.feedbackQuestions.get("qn1ForSecondSession");
+        questionDetails = (FeedbackMsqQuestionDetails) copiedQuestion.getQuestionDetailsCopy();
+        feedbackEditPage.copyQuestion(copiedQuestion.getCourseId(),
+                copiedQuestion.getQuestionDetailsCopy().getQuestionText());
+        copiedQuestion.setFeedbackSession(feedbackSession);
+        copiedQuestion.setQuestionNumber(3);
+
+        feedbackEditPage.verifyMsqQuestionDetails(3, questionDetails);
+        verifyPresentInDatabase(copiedQuestion);
+
+        ______TS("edit question");
+        questionDetails = (FeedbackMsqQuestionDetails) loadedQuestion.getQuestionDetailsCopy();
+        questionDetails.setHasAssignedWeights(false);
+        questionDetails.setMsqWeights(new ArrayList<>());
+        questionDetails.setOtherEnabled(false);
+        questionDetails.setMsqOtherWeight(0);
+        questionDetails.setMaxSelectableChoices(Const.POINTS_NO_VALUE);
+        List<String> choices = questionDetails.getMsqChoices();
+        choices.add("Edited choice");
+        questionDetails.setMsqChoices(choices);
+        loadedQuestion.setQuestionDetails(questionDetails);
+        feedbackEditPage.editMsqQuestion(2, questionDetails);
+        feedbackEditPage.waitForPageToLoad();
+
+        feedbackEditPage.verifyMsqQuestionDetails(2, questionDetails);
+        verifyPresentInDatabase(loadedQuestion);
+    }
+
+    @Override
+    protected void testSubmitPage() {
+        FeedbackSubmitPage feedbackSubmitPage = loginToFeedbackSubmitPage();
+        testData = loadSqlDataBundle("/FeedbackMsqQuestionE2ETest_SqlEntities.json");
```

Comment:
Do we need to reload the test data? I thought using the previous test data would be fine?

Reply:
https://github.com/TEAMMATES/teammates/blob/61d89d4b113820ebea7751a98cdb4ef73b513195/src/e2e/java/teammates/e2e/cases/sql/FeedbackMsqQuestionE2ETest.java#L82

It was because this changes the objects in `testData`. It is probably more efficient to make a copy, I have changed it to use a copy instead.